godot/src/config/config_data.gd

```diff
@@ -116,6 +120,38 @@ var audio_general_volume: float = 100.0:
 		param_changed.emit(ConfigParams.AUDIO_GENERAL_VOLUME)
 
 
+func fix_last_places_duplicates(place_dict: Dictionary, _last_places: Array):
+	var realm = place_dict.get("realm")
+	var position = place_dict.get("position")
+	var to_remove: Array[int] = []
+	for i in range(_last_places.size()):
+		var place = _last_places[i]
+		var place_realm = place.get("realm")
+		var place_position = place.get("position")
+		if place_realm == realm:
+			if Realm.is_genesis_city(realm):
+				if place_position == position:
+					to_remove.push_front(i)
+			else:
+				to_remove.push_front(i)
+
+	for i in to_remove:
```

Comment:
This works because of the way items are pushed into the vector.
Add a comment or use by_reference remove (`to_remove.push(_last_places[i])` and then use erase). Don't think the performance makes a big difference here